ansibullbot/triagers/plugins/component_matching.py

```diff
@@ -0,0 +1,224 @@
+                )
+
+                # keep track of namespace maintainers for new mods too
+                ns = match['namespace']
+                match['namespace_maintainers'] = \
+                    module_indexer.get_maintainers_for_namespace(ns)
+
+                # these are "community" supported from the beginning?
+                match['metadata']['supported_by'] = 'community'
+
+                cmeta['module_match'] = copy.deepcopy(match)
+                cmeta['component'] = match['name']
+
+            elif f.endswith('.md'):
+                # network/avi/README.md
+                continue
+            else:
+                # FIXME - what do with these files?
+                logging.warning('unhandled filepath for matching: %s' % f)
+
+    # get labels for files ...
+    if not iw.is_pullrequest():
+        cmeta['is_issue'] = True
+        cmeta['is_pullrequest'] = False
+        cmeta['component_labels'] = []
+
+        if not cmeta['is_module']:
+            components = file_indexer.find_component_match(
+                iw.title,
+                iw.body,
+                iw.template_data
+            )
+            cmeta['guessed_components'] = components
+            if components:
+                comp_labels = file_indexer.get_component_labels(
+                    valid_labels,
+                    components
+                )
+                cmeta['component_labels'] = comp_labels
+            else:
+                cmeta['component_labels'] = []
+
+    else:
+        cmeta['is_issue'] = False
+        cmeta['is_pullrequest'] = True
+        cmeta['component_labels'] = \
+            file_indexer.get_component_labels(
+                iw.files,
+                valid_labels=valid_labels
+            )
+
+    # who owns this? FIXME - is this even used?
+    cmeta['owner'] = 'ansible'
+    if cmeta['module_match']:
+        print(cmeta['module_match'])
+        maintainers = cmeta['module_match']['maintainers']
+        if maintainers:
+            cmeta['owner'] = maintainers
+        elif cmeta['is_new_module']:
+            cmeta['owner'] = ['ansible']
+        else:
+            logging.error('NO MAINTAINER LISTED FOR %s'
+                          % cmeta['module_match']['name'])
+
+    elif cmeta['is_pullrequest'] and cmeta['component_matches']:
+        cmeta['owner'] = []
+        for cmatch in cmeta['component_matches']:
+            for user in cmatch['owners']:
+                if user not in cmeta['owner']:
+                    cmeta['owner'].append(user)
+            for user in cmatch['notify']:
+                if user not in cmeta['owner']:
+                    cmeta['owner'].append(user)
+
+    # if any component is core, then the whole thing is core
+    if cmeta['component_matches']:
+        core = False
+        for cmatch in cmeta['component_matches']:
+            if cmatch.get('supported_by') in ['networking', 'core']:
+                core = True
```

Comment:
Either the comment above is wrong, or a `break` statement is missing.
Note that if the comment is right, `any(cmatch.get('supported_by') in ['networking', 'core'] for cmatch in cmeta['component_matches'])` could be used.